crypto/x509/x509_vfy.c

```diff
@@ -327,14 +328,11 @@ static X509 *find_issuer(X509_STORE_CTX *ctx, STACK_OF(X509) *sk, X509 *x)
 
     for (i = 0; i < sk_X509_num(sk); i++) {
         issuer = sk_X509_value(sk, i);
-        /*
-         * Below check 'issuer != x' is an optimization and safety precaution:
-         * Candidate issuer cert cannot be the same as the subject cert 'x'.
-         */
-        if (issuer != x && ctx->check_issued(ctx, x, issuer)
+        if (ctx->check_issued(ctx, x, issuer)
             && (((x->ex_flags & EXFLAG_SI) != 0 && sk_X509_num(ctx->chain) == 1)
                 || !sk_X509_contains(ctx->chain, issuer))) {
-            rv = issuer;
+            if (rv == NULL)
```

Comment:
Why this if?

Reply:
This is a minor "drive-by" improvement also mentioned in the adapted comment internally documenting the function:
> Prefer the first one that is not expired, else take the **first** expired one.

Before, it preferred the **last** expired one, which would be unexpected and inconsistent with other (partly documented) lookup behavior.

Reply:
But below you are calling x509_check_cert_time on the rv and not on the issuer. This looks wrong.

Reply:
Oh, indeed! Before the change it did not matter, but now it does.
Fixed.